srcs/vm/get_arguments.c

```diff
@@ -0,0 +1,40 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        ::::::::            */
+/*   get_arguments.c                                    :+:    :+:            */
+/*                                                     +:+                    */
+/*   By: merlijn <merlijn@student.codam.nl>           +#+                     */
+/*                                                   +#+                      */
+/*   Created: 2020/08/12 14:41:11 by merlijn       #+#    #+#                 */
+/*   Updated: 2020/08/13 15:30:38 by merlijn       ########   odam.nl         */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "vm.h"
+
+/*
+**	GET_ARGUMENT_TYPES
+**	Uses bitwise operators to split the encoding byte
+**	into all three types
+**	T_REG = 0x01 = 1
+**	T_DIR = 0x10 = 2
+**	T_IND = 0x11 = 3
+**	empty = 0x00 = 0
+*/
+
+void	get_argument_types(char *mem, t_cursor *cursor)
+{
+	int	i;
+	int byte;
+
+	byte = mem[(cursor->pos + 1) % MEM_SIZE];
```

Comment:
Waarom wordt hier + 1 gedaan?

Reply:
Arg is toch achter de cursor?

Reply:
cursor->pos byte is de opcode, cursor->pos + 1 is de encoding byte